salt/transport/tcp.py

```diff
@@ -764,6 +763,42 @@ def _create_stream(self, max_buffer_size, af, addr, **kwargs):  # pylint: disabl
         return stream.connect(addr)
 
 
+class SaltMessageClientPool(salt.transport.MessageClientPool):
+    '''
+    Wrapper class of SaltMessageClient to avoid blocking waiting while writing data to socket.
+    '''
+    def __init__(self, opts, args=None, kwargs=None):
+        super(SaltMessageClientPool, self).__init__(SaltMessageClient, opts, args=args, kwargs=kwargs)
+
+    def __del__(self):
+        self.close()
+
+    def close(self):
+        for message_client in self.message_clients:
+            message_client.close()
+
```

Comment:
Should there be a:

```
self.message_clients = []
```

here in case `close` is called more than once (this may be a good practice even though each individual client is tolerant to having its `close` called more than once so this would work as is).